accepted/0000-replace-dwr.md

```diff
@@ -0,0 +1,74 @@
+
+  <sup>*</sup> All of these are only used in the [Your RHN](https://github.com/uyuni-project/uyuni/blob/master/java/code/webapp/WEB-INF/pages/yourrhn.jsp) page.
+
+## What is the expected outcome?
+SUMA working normally with the same AJAX requests being made and handled without any use of the DWR library.
+
+
+# Detailed design
+
+Although the solution could be a little risky, since it involves changes in a number of stable features, it is not so complex. The idea is to have a raw Servlet incorporated in SUMA working as backend for all the AJAX requests currently handled by the [DwrServlet](https://github.com/uyuni-project/uyuni/blob/master/java/code/webapp/WEB-INF/web.xml#L260). This servlet could be based on the URLs of the requests to determine which Java class to instantiate and which method to call on it.
+
+For the frontend side, the capabilities provided by DWR (both the utility functions from `util.js` file and the ajax calls) are replaceable using jQuery.
+
+There is a PoC in [this commit](https://github.com/wweellddeerr/uyuni/commit/9ddb68030dc49fa3272a062ceb7015c35b68d457) exploring how it is possible to use jQuery to make the ajax request and a raw Servlet as backend replacing the `DWRItemSelector` component of DWR. The SSM (one of the features that uses it) seems to work well after applying this patch.
+
+Using a raw Servlet as backend seems to be the best approach since (a) the methods currently exposed by DWR are only available through the `DwrServlet`, which is a raw servlet; and (b) these methods typically manipulate the HttpSession, making it difficult to use the HTTP API, for example.
+
+# Drawbacks
+
+Why should we **not** do this?
+
+Except for the [browser warning regarding DWRSESSIONID cookie](https://github.com/SUSE/spacewalk/issues/13995) that seems to be urgent, as it's likely that the requests are going to be rejected in some future version updates, there is no clear evidence of major problems using DWR and, in fact, it has been used for a long time in SUMA and seems to be stable.
+
+# Alternatives
+
+The first alternative is to just keep using DWR and try to solve the warning mentioned above, maybe trying to create a Pull Request on the DWR repository or even opening an issue on it.
+
+The second alternative is to gradually replace the pages using DWR to use React. The main drawbacks of this alternative are: (1) some issues to migrate pages using DWR were aborted in the past, probably due to non technical decisions; and (2) the `rl:selectablecolumn` custom tag mentioned before is used in several JSP pages, it would be necessary to find an alternative to this component in order to avoid the need to migrate so many pages.
+
+A hybrid approach could also be used, migrating some pages (specially the `Your RHN`) to React and applying the suggested solution (or even the first alternative) only in the remaining cases, reducing the risks.
```

Comment:
Ultimately we will want to port all pages to React, but I think the subgoal of getting rid of DWR is an important one as well.

Please use your best judgement to select the cheapest approach - if migrating a page to React is easier than adapting the corresponting DWR methods, please go down that path.